plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -48,15 +66,23 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
    std::optional<state_history_log> trace_log;
    std::optional<state_history_log> chain_state_log;
    bool                             trace_debug_mode = false;
-   bool                             stopping         = false;
+   std::atomic<bool>                stopping         = false;
    std::optional<scoped_connection> applied_transaction_connection;
    std::optional<scoped_connection> block_start_connection;
    std::optional<scoped_connection> accepted_block_connection;
-   string                           endpoint_address = "0.0.0.0";
+   string                           endpoint_address;
    uint16_t                         endpoint_port    = 8080;
    std::unique_ptr<tcp::acceptor>   acceptor;
```

Comment:
Convert `acceptor` to an `std::variant<tcp::acceptor, unixs::acceptor>` and use their common API whenever possible.